### PR TITLE
Implements FromGlibPtrFull<*const T> for boxed types.

### PR DESCRIPTION
**Pull request by luukvanderduim**
The aim is to solve issue: 865 of the gir repository [865](https://github.com/gtk-rs/gir/issues/865)

AnyBox::Native(Box<T>) looks like the suitable variant for a fully owned *const T
Is checking non_null sufficient here before converting the raw pointer to a box?
I'm assuming here MM::copy() assures other type requirements are met.
Is this correct? 
While this solution makes my code compile and most of this impl is copied and adapted from other impls in the same file, there may well be things I have missed or misunderstood. Therefore a close look at what I've done is much appreciated.

### CODE REVIEW

File: src/boxed.rs [hunk 1]
```diff
@@ -457,6 +466,18 @@ impl<T: 'static, MM: BoxedMemoryManager<T>> FromGlibPtrFull<*mut T> for Boxed<T,
     }
 }
 
+impl<T: 'static, MM: BoxedMemoryManager<T>> FromGlibPtrFull<*const T> for Boxed<T, MM> {
+    #[inline]
+    unsafe fn from_glib_full(ptr: *const T) -> Self {
+        assert!(!ptr.is_null());
+        let ptr = MM::copy(ptr);
+        Boxed {
+            inner: AnyBox::Native(Box::from_raw(ptr)),
```
Comment (sdroege) [inline src/boxed.rs]: This is only allowed if the pointer was originally a `Box` and allocated from Rust. That's not the case here :)

Also simply copying the memory behind the pointer is not enough here. You'd leak memory.

What you want here is to call the `*mut _` version of `from_glib_full()` by simply casting the pointer.

Comment (luukvanderduim) [reply src/boxed.rs]: Thanks! Will try to get that in place.

Comment (luukvanderduim) [reply src/boxed.rs]: Simplest proposal:
`unsafe fn from_glib_full(ptr: *const T) -> Self {
        assert!(!ptr.is_null());
        let ptr = ptr as *mut _;
        Boxed {
            inner: AnyBox::ForeignOwned(ptr::NonNull::new_unchecked(ptr)),
            _dummy: PhantomData,
        }
    }`
This is essentially the same as the FromGlibPtrFull on *mut but with cast to *mut. Would this suffice?

Comment (sdroege) [reply src/boxed.rs]: You can do it even easier, but basically yes. You could do `from_glib_full(ptr as *mut T)`

Comment (luukvanderduim) [reply src/boxed.rs]: Could we even generalize over *mut/*const pointers by using a suitable pointer trait in the signature? I quickly tried Ptr but run into trouble.

Comment (sdroege) [reply src/boxed.rs]: I don't think so

**Comment by luukvanderduim** [comment]
Travis'CI has an osx-nightly run that fails. I've seen this yesterday  as well. 
```
Caused by:
  Unable to update https://github.com/gtk-rs/gio 
```
Harmless Travis quirk?

**Comment by EPashkin** [comment]
Restarted job

**Pull request merged**